tests/chains/Cosmos/THORChain/SwapTests.cpp

```diff
@@ -761,6 +763,60 @@ TEST(THORChainSwap, SwapBnbRune) {
     // https://viewblock.io/thorchain/tx/D582E1473FE229F02F162055833C64F49FB4FF515989A4785ED7898560A448FC
 }
 
+TEST(THORChainSwap, SwapBtcCacao) {
+    Proto::Asset fromAsset;
+    fromAsset.set_chain(static_cast<Proto::Chain>(Chain::BTC));
+    Proto::Asset toAsset;
+    toAsset.set_chain(static_cast<Proto::Chain>(Chain::MAYA));
+    toAsset.set_symbol("CACAO");
+    auto&& [out, errorCode, error] = SwapBuilder::builder()
+                                         .from(fromAsset)
+                                         .to(toAsset)
+                                         .fromAddress(Address1Btc)
+                                         .toAddress(Address1Maya)
+                                         .vault(VaultBtc)
+                                         .fromAmount("4000000")
+                                         .toAmountLimit("121065076")
+                                         .fromChain(Chain::MAYA)
```

Comment:
I guess `fromChain` is only needed to specify that MAYASwap should be used instead of THORSwap.
I think there should be another way.
So if MAYASwap is only used to exchange `MAYA(CACAO)` (sell or buy), then you can check if either `from` or `to`  is MAYA chain.

Reply:
The problem is that `MAYASwap` can be use not only for exchanging `CACAO`, it can also exchange `ASSET A` to `ASSET B`. Also, in the case where we want to SWAP RUNE -> CACAO or viceversa I think the `fromChain` should work better, what do you think?